python-packages/presto_viewmodels_drf/presto_viewmodels_drf/serializers.py

```diff
@@ -0,0 +1,46 @@
+class OptinFieldsSerializerMixin:
+    """
+    Regulates fields exposed by default & as requested.
+
+    Construct URLs like /?include_fields=first_name,email to opt-in into fields.
+
+    By default, all "opt_in_only_fields" are excluded from serializer, unless specifically opted-in.
+
+    If any of fields specified to be included does not belong to opt_in_only_fields, the serializer
+    will perform in "strict" mode that ONLY fields specced are included.
+    """
+
+    def __init__(self, *args, **kwargs):
+        result = super().__init__(*args, **kwargs)
+
+        # only if context.request is available (ie. not during inspection etc)
+        if hasattr(self, "context") and (
+            "request" in self.context or "include_fields" in self.context
+        ):
+            heavy_fields = set(getattr(self.Meta, "opt_in_only_fields", []))
+
+            if "include_fields" in self.context["request"].query_params:
```

Comment:
this will break if no 'request'

(which highlights you are missing a test for `context["include_fields"]`)

Reply:
👏